slides/week0/week0.org

```diff
@@ -0,0 +1,426 @@
+    }
+  }
+#+END_SRC
+
+* Conditional Practice 1 Question
+#+BEGIN_NOTES
+1
+#+END_NOTES
+- 1 or 2?
+#+BEGIN_SRC C++ -n
+if (2 == 2) {
+    // 1
+} else {
+    // 2
+}
+#+END_SRC
+
+* Conditional Practice 2 Question
+#+BEGIN_NOTES
+2
+#+END_NOTES
+- 1 or 2?
+#+BEGIN_SRC C++ -n
+if (2 != 2) {
+    // 1
+} else {
+    // 2
+}
+#+END_SRC
+
+* Conditional Practice 3 Question
+#+BEGIN_NOTES
+2
+#+END_NOTES
+- 1 or 2?
+#+BEGIN_SRC C++ -n
+if (2 < 2) {
+    // 1
+} else {
+    // 2
+}
+#+END_SRC
+
+* Chaining Conditionals
+#+BEGIN_NOTES
+Talk about how this is difficult to write at deeper levels
+Think 5-6 Conditionals
+#+END_NOTES
+#+BEGIN_SRC C++ -n
+  int main() {
+    if (true) {
+        // <==== This executes =====
+        if (true) {
+            // <==== This executes =====
+        }
+        // <==== This executes =====
+    }
+    // <==== This executes =====
+  }
+#+END_SRC
+
+* Logical Operators *AND*
+- *&&*
+| a | b | a AND b|
+|-------+------|------+------|
+| True | True | True |
+| True | False | False |
+| False | True | False |
+| False | False | False |
+
+* Logical Operators *OR*
+- *||*
+| a | b | a OR b|
+|-------+------|------+------|
+| True | True | True |
+| True | False | True |
+| False | True | True |
+| False | False | False |
+
+* Logical Operators order
+- statements will be evaluated from left to right
+- no limit to the number you can have
+- Statements can be grouped using *()*
+    - Just like with math operators *()* are executed first
+    - 1 && (2 || 3)
+        - 2 || 3 is done first and the result in anded with 1
+
+* Conditional Practice 4 Question
+#+BEGIN_NOTES
+2
+#+END_NOTES
+- 1 or 2?
+#+BEGIN_SRC C++ -n
+if (2 < 2 && 2 == 2) {
+    // 1
+} else {
+    // 2
+}
+#+END_SRC
+
+* Conditional Practice 4 Question
+#+BEGIN_NOTES
+1
+#+END_NOTES
+- 1 or 2?
+#+BEGIN_SRC C++ -n
+if (2 != 2 || 2 == 2) {
+    // 1
+} else {
+    // 2
+}
+#+END_SRC
+
+* Loops!
+- What If I want to do something multiple times
+
+* While Loops Syntax
+#+BEGIN_SRC C++ -n
+while (condition == true) {
+    // do something
+}
+#+END_SRC
+
+* For Loop Syntax
+#+BEGIN_SRC C++ -n
+for (initializer; condition; change condition) {
+    // do something
+}
+#+END_SRC
+
+* Math Operators Continued
+| syntax | meaning | equivalent |
+|-------+------|------+------|
+| =var++;= | =increments the variable= | var = var + 1; |
+| =var--;= | =decrements the variable= | var = var - 1; |
+
+* Equivalent while and for Loop
+#+BEGIN_SRC C++ -n
+int while_counter = 10;
+while (while_counter > 0) {
+    // do something
+    while_counter = while_counter - 1;
+}
+for (int for_counter = 10; for_counter > 0; for_counter++) {
+    // do something
+}
+#+END_SRC
+
+* Questions??
+- A go-to place to ask questions ([[https://piazza.com/gatech/fall2017/rjsw/home][Piazza!]])
+- Monitored by highly experienced RoboJackets software members
+
+* Let's write some code
+#+BEGIN_NOTES
+Write a couple examples of code using what has already been covered.
```

Comment:
I think it might be a little safer to add a few examples here, not really a blocker though, just more that we should plan it before we actually go for it.